src/coreclr/vm/dllimport.cpp

```diff
@@ -2105,11 +2105,11 @@ void NDirectStubLinker::DoNDirect(ILCodeStream *pcsEmit, DWORD dwStubFlags, Meth
                 // pcsEmit->EmitCALL(METHOD__STUBHELPERS__GET_NDIRECT_TARGET, 1, 1);
 
 #ifdef _DEBUG // There are five more pointer values for _DEBUG
-                _ASSERTE(offsetof(NDirectMethodDesc, ndirect.m_pWriteableData) == sizeof(void*) * 8 + 8);
-                pcsEmit->EmitLDC(TARGET_POINTER_SIZE * 8 + 8); // offsetof(NDirectMethodDesc, ndirect.m_pWriteableData)
+                _ASSERTE(offsetof(NDirectMethodDesc, ndirect.m_pWriteableData) == sizeof(void*) * 7 + 8);
+                pcsEmit->EmitLDC(TARGET_POINTER_SIZE * 7 + 8); // offsetof(NDirectMethodDesc, ndirect.m_pWriteableData)
```

Comment:
@jkotas, any reason why we have this manually rather than using `offsetof`?

It looks like `TARGET_POINTER_SIZE` is documented as matching `sizeof(void*)` elsewhere and `offsetof` is evaluated to be a constant by the compiler.

Reply:
We used to have (or at least experimented with) cross-bitness NGen in .NET Framework at one point. This looks like a left-over from that.

Reply:
It should be fine to switch these to regular `offsetof`

Reply:
Thanks! Updated.

Notably I'd have expected maybe a slightly different way of supporting cross-bitness so that things could still be typesafe if we were doing that.